File: lib/features/department/models/department.dart
// 部门 model（Phase 2）
// 文档：docs/04-数据模型/实体字典.md#department

class Department {
  const Department({
    required this.id,
    required this.code,
    required this.name,
    this.parentId,
    this.managerName,
    this.sortOrder = 0,
    this.headcount = 0,
    this.children = const [],
  });

  final String id;
  final String code;
  final String name;
  final String? parentId;
  final String? managerName;
  final int sortOrder;
  final int headcount;
  final List<Department> children;
}

File: server/src/main/java/com/uten/imp/features/org/department/Department.java
package com.uten.imp.features.org.department;

import com.uten.imp.common.domain.SoftDeletableEntity;
import com.uten.imp.features.org.employee.Employee;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 部门（组织架构树）。邻接表 parent + 语义 level + 物化路径 path（path 由 DB 触发器维护）。
 */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "departments")
public class Department extends SoftDeletableEntity {

    @Column(nullable = false, unique = true)
    private String code;

    @Column(nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Department parent;

    @Column(nullable = false)
    private String level;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manager_id")
    private Employee manager;

    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    @Column(nullable = false)
    private String path = "/";

    @Column(nullable = false)
    private Integer headcount = 0;
}
